lib/scarlet/models/nick.rb

```diff
@@ -1,25 +1,50 @@
 require 'scarlet/models/model_base'
+require 'moon-repository/query'
 
 class Scarlet
-  # Privileges level
-  # 0 - regular
-  # 1 - registered
-  # 2 - voice
-  # 3 - VIP
-  #...
-  # 6 - super tester
-  # 7 - op
-  # 8 - dev
-  # 9 - owner
   class Nick < ModelBase
     field :nick,       type: String,  validate: { presence: {} }
     field :aliases,    type: Array,   default: proc { Array.new }
-    field :privileges, type: Integer, default: 1
+    # owner > dev > admin > mod > (every other user)
+    field :groups,     type: Array,   default: proc { ['registered'] }
     field :win_points, type: Integer, default: 0
     field :settings,   type: Hash,    default: proc { Hash.new }
 
+    # Checks if the nick is apart of the given groups
+    #
+    # @param [String] expected
+    # @return [Boolean]
+    def groups?(*expected)
+      (groups & expected) == expected
+    end
+    alias :group? :groups?
+
+    def owner?
+      group?('owner')
+    end
+
+    def dev?
+      owner? || group?('dev')
+    end
+
+    def admin?
+      dev? || group?('admin')
+    end
+
+    def mod?
+      admin? || group?('mod')
```

Comment:
I elect getting rid of most of these. devs, mods, admins... Let's say there's a `sudo`/`gawd`/`staff` group with full 'root' access, and a `user`/`registered` group for normal users. Then each of the plugins add topical groups as needed (same way linux does so with audio/printer/...), so for instance we add a `polyfox` group. If we decide to widen access to eval, we could potentially add `eval` groups, but right now, the lockdown is insufficient, so 2-3 groups total should be enough.